### PR TITLE
gtk+ test cleanup

### PR DESCRIPTION
**Pull request by tschoonj**
Part of the tasks set out in #39584

### CODE REVIEW

File: Library/Formula/gtk+.rb [hunk 1]
```diff
@@ -45,7 +46,60 @@ def install
...
+    ENV.prepend "PKG_CONFIG_PATH", OS::Mac::X11.lib/"pkgconfig"
```
Comment (xu-cheng) [inline Library/Formula/gtk+.rb]: This has no effect. It's not used by any logic in below. Can you try to remove it and test whether it works without this.

**Comment by tschoonj** [comment]
No, the line is essential: otherwise it won't find the X11 headers.

It's basically the equivalent of ENV.x11 but that won't pass `audit --strict`

I made a comment about this in #39578

**Comment by xu-cheng** [comment]
You never used `PKG_CONFIG_PATH` variable as you never run `pkg-config` in your test.

**Comment by tschoonj** [comment]
Good point! In that case the ENV.x11 is completely redundant too.

**Comment by xu-cheng** [comment]
You can go to have a try. Otherwise, it means it should `depends_on :x11`

**Comment by tschoonj** [comment]
Ok removed the line both in gtk+ and gtkmm

By the way: you cannot add `depends_on` inside of test blocks: run time error.

**Comment by xu-cheng** [comment]
> By the way: you cannot add depends_on inside of test blocks: run time error.

Actually I mean this test depends on x11. So either you find another test which doesn't depend on x11, or you should add `depends_on :x11` into the dependencies list. This will also mean this formula should be moved to `homebrew/x11`.

**Comment by tschoonj** [comment]
Well I think that we'll soon make gtk+, pango and cairo quartz only (#38969) to end the nightmare that is supporting both x11 and quartz backends for gtk+. gtk3 doesn't even support x11 anymore starting with version 3.16.0.

**Comment by DomT4** [comment]
Moving GTK+ to Homebrew/x11 will be mayhem; so much depends on it it's not really feasible as an outright move. `depends_on :x11` is already recommended in this formula, and most `test do` blocks only support the default formula compile and settings, so relying on `x11` to be present for the test isn't a major sin for now. The end goal is to remove the x11 backend, but till then :neutral_face:.

**Pull request closed without merging**